engines/gob/inter_v7.cpp

```diff
@@ -619,40 +980,437 @@ void Inter_v7::o7_getDBString() {
+		// We skip writing the file size into VAR(16) here as a workarond (the value is not used anyway).
+	} else
+		WRITE_VAR(16, (uint32) size);
 }
 
+void Inter_v7::o7_readData(OpFuncParams &params) {
+	Common::String file = getFile(_vm->_game->_script->evalString(), false);
+
+	uint16 dataVar = _vm->_game->_script->readVarIndex();
+	int32  size    = _vm->_game->_script->readValExpr();
+	int32  offset  = _vm->_game->_script->evalInt();
+	int32  retSize = 0;
+
+	debugC(2, kDebugFileIO, "Read from file \"%s\" (%d, %d bytes at %d)",
+		   file.c_str(), dataVar, size, offset);
+
+	SaveLoad::SaveMode mode = _vm->_saveLoad->getSaveMode(file.c_str());
+	if (mode == SaveLoad::kSaveModeSave) {
+
+		WRITE_VAR(1, 1);
+
+		if (!_vm->_saveLoad->load(file.c_str(), dataVar, size, offset)) {
+			// Too noisy, the scripts often try to load "save" file not existing (yet)
+			// GUI::MessageDialog dialog(_("Failed to load saved game from file."));
+			// dialog.runModal();
+		} else
+			WRITE_VAR(1, 0);
+
+		return;
+
+	} else if (mode == SaveLoad::kSaveModeIgnore)
+		return;
+
+	if (size < 0) {
+		if (readSprite(file, dataVar, size, offset))
+			WRITE_VAR(1, 0);
+		return;
+	} else if (size == 0) {
+		dataVar = 0;
+		size = _vm->_game->_script->getVariablesCount() * 4;
+	}
+
+	byte *buf = _variables->getAddressOff8(dataVar);
+
+	if (file.empty()) {
+		WRITE_VAR(1, size);
+		return;
+	}
+
+	WRITE_VAR(1, 1);
+	Common::SeekableReadStream *stream = _vm->_dataIO->getFile(file);
+	if (!stream)
+		return;
+
+	_vm->_draw->animateCursor(4);
+	if (offset > stream->size()) {
+		warning("oPlaytoons_readData: File \"%s\", Offset (%d) > file size (%d)",
+				file.c_str(), offset, (int)stream->size());
+		delete stream;
+		return;
+	}
+
+	if (offset < 0)
+		stream->seek(offset + 1, SEEK_END);
+	else
+		stream->seek(offset);
+
+	if (((dataVar >> 2) == 59) && (size == 4)) {
+		WRITE_VAR(59, stream->readUint32LE());
+		// The scripts in some versions divide through 256^3 then,
+		// effectively doing a LE->BE conversion
+		if ((_vm->getPlatform() != Common::kPlatformDOS) && (VAR(59) < 256))
+			WRITE_VAR(59, SWAP_BYTES_32(VAR(59)));
+	} else
+		retSize = stream->read(buf, size);
+
+	if (retSize == size)
+		WRITE_VAR(1, 0);
+
+	delete stream;
+}
+
+void Inter_v7::o7_writeData(OpFuncParams &params) {
+	Common::String file = getFile(_vm->_game->_script->evalString(), false);
+
+	int16 dataVar = _vm->_game->_script->readVarIndex();
+	int32 size    = _vm->_game->_script->readValExpr();
+	int32 offset  = _vm->_game->_script->evalInt();
+
+	debugC(2, kDebugFileIO, "Write to file \"%s\" (%d, %d bytes at %d)",
+		   file.c_str(), dataVar, size, offset);
+
+	WRITE_VAR(1, 1);
+
+	SaveLoad::SaveMode mode = _vm->_saveLoad ? _vm->_saveLoad->getSaveMode(file.c_str()) : SaveLoad::kSaveModeNone;
+	if (mode == SaveLoad::kSaveModeSave) {
+
+		if (!_vm->_saveLoad->save(file.c_str(), dataVar, size, offset)) {
+
+			GUI::MessageDialog dialog(_("Failed to save game to file."));
```

Comment:
Since you added a translatable string here, please also add `engines/gob/inter_v7.cpp` to `engines/gob/POTFILES`